csrc/scheduler/multi_matmul.cpp

```diff
@@ -1296,22 +1294,292 @@ class MultipleMatmulScheduler {
+    // [Mo, No, m*n/vect/TIDx/TIDy/TIDz, TIDz, TIDy, TIDx, vect]
+
+    for (TensorView* mma_result : mma_results_) {
+      // step-5, Parallel first 2 dims same as mma_result
+      scheduler_utils::parallelizeAllLike(
+          mma_result,
+          2,
+          {c},
+          {ParallelType::BIDx, ParallelType::BIDy, ParallelType::BIDz});
+    }
+  }
+
   void scheduleEpilogue() {
-    NVF_THROW("scheduleEpilogue is not yet implemented");
+    std::vector<TensorView*> output_tvs;
+    for (Val* v : fusion_->outputs()) {
+      if (auto tv = dynamic_cast<TensorView*>(v)) {
+        output_tvs.push_back(tv);
+      }
+    }
+    if (params_->use_smem_epilogue) {
+      blockTileTensors(output_tvs);
+      for (auto [dc, d] : cached_outputs_) {
+        // Schedule output tensor differently for better global memory access
+        // pattern.
+        scheduleOutputTensor(d);
+        d->axis(-1)->parallelize(ParallelType::Vectorize);
+
+        // Propagate output tensor transformations back to smem_epilogue
+        scheduler_utils::BoundedDirectionalTransformPropagator::backward(
+            d, -1, smem_epilogues_);
+      }
+    } else {
+      for (TensorView* mma_result : mma_results_) {
+        scheduler_utils::BoundedDirectionalTransformPropagator::forward(
+            mma_result,
+            -1,
+            output_tvs,
+            scheduler_utils::BoundedDirectionalTransformPropagator::Options()
+                .propagateParallelType()
+                .propagateToBoundary());
+      }
+      for (auto [dc, d] : cached_outputs_) {
+        // We might propagate an inner dimension that is not compatible with the
+        // output or bias-like inputs. In those cases, we will further split
+        // this dimension with an outer unrolled loop to achieve the proper
+        // vectorization as specified by params.supported_vec_size.epilogue.
+        NVF_ERROR(d->axis(-1)->extent()->isConst());
+        int64_t d_extent = d->axis(-1)->extent()->value().as<int64_t>();
+        if (d_extent > params_->supported_vec_size.epilogue) {
+          // Should always be a divisible split
+          NVF_ERROR(d_extent % params_->supported_vec_size.epilogue == 0);
+          d->split(
+              -1, params_->supported_vec_size.epilogue, /*inner_split=*/true);
+          d->axis(-2)->parallelize(ParallelType::Unroll);
+        }
+        d->axis(-1)->parallelize(ParallelType::Vectorize);
+      }
+    }
+
+    // propagate output transformations to all inputs that are part of epilogue
+    //  operations, input tvs with non-core roles
+    //  core roles: essential for matmul, for example mma inputs' producers
+    scheduleFusionInputsForEpilogue();
+  }
+
+  //! Propagates transformations from fusion output to fusion tv inputs that are
+  //!  producers in the epilogue. Transformations' propagation aims at input tvs
+  //!  which are not assigned to core roles, that is, are not MMA inputs.
+  void scheduleFusionInputsForEpilogue() {
```

Comment:
This is copied verbatim from matmul.cpp, other than reformatting a comment and using `params_->use_smem_epilogue` instead of a function argument.